Automattic/jetpack
# Blocks: Show the Launchpad on save modal inside the Site Editor for all flows

Fixes [LaunchPad - Site Editor Save Modal - Adapt for non-LIB flows](https://github.com/Automattic/wp-calypso/issues/71717) and [LaunchPad - editor 'back-to-launchpad' modal tracking events](https://github.com/Automattic/wp-calypso/issues/72097)

## Proposed changes:
This PR provides a generic modal for all Launchpad flows to allow the user to go back to the Launchpad screen from the Site Editor once they have saved their progress. It also adds tracking events

<img width="866" alt="image" src="https://user-images.githubusercontent.com/20927667/213217851-efc777e5-b74a-4b00-8219-75dea44cd11c.png">

### Other information:


## Jetpack product discussion
paYE8P-1Kg-p2

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

### Testing on WPCOM
1. Sandbox `public-api.wordpress.com` along with a Launchpad enabled site
2. Run `bin/jetpack-downloader test jetpack add/launchpad_save_modal_all_flows` on your sandbox
3. Go to the Site Editor
4. Make a change and save. You should see the modal appear.
5. Test with different flows - `newsletter`, `link-in-bio`, `free`, `videopress`.
6. Use [Tracks Vig ](https://github.com/Automattic/tracks-chrome-extension) to test the tracks events:
• Clicking the close button -> `jetpack_launchpad_save_modal_close`
• Clicking the Back to Edit -> `jetpack_launchpad_save_modal_back_to_edit`
• Clicking the Next Button -> `jetpack_launchpad_save_modal_next_steps`
